### ISSUE TITLE
Implement automated E2E testing of alerts

### ISSUE THREAD

**Issue by nickbp** [opstrace/opstrace#442]
Once #430 is complete and it looks like alerts are working, we should have something that automates setting up some alerts and checking that they work. The cortex alertmanager/ruler seem to internally still be in flux so it'd be good to immediately catch regressions as we update cortex over time. Alerts in particular are pretty sensitive - they really need to work consistently.

Seb pointed me to https://github.com/grafana/cortex-tools/blob/main/docs/e2ealerting.md which may be useful for this?

**Comment by triclambert** [comment]
Moving this to backlog, as a lower priority than the blog post and user guide.

**Comment by nickbp** [comment]
Implemented in #547

**Comment by nickbp** [comment]
Quick status update since it's been a while:

The implementation (in aforementioned PR #547 ) involved adding a fair bit of functionality (e.g. the opstrace/kubernetes library) into test-remote, but that should be useful for other similar tests in the future. As a bonus the test ends up validating not only that the alertmanager/ruler configs are working, but also that per-tenant prometheus scraping and routing metrics into cortex is working.

Following initial implementation working when driven from my workstation, there was a fair bit of work getting the E2E tests to build/run in CI with the added dependencies, but that seems to be all set now. At this point the E2E tests have been consistently passing, with one flake discovered with waiting for the e2ealerting pod to be discovered by the tenant prometheus for scraping sometimes taking longer than the timeout.

So at this point the test should be ready to go, but I may give it another couple runs to catch any additional flakes before merging.

**Comment by nickbp** [comment]
(Didn't see any additional flakes caused by this test, so :ship: )
